deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+    ) {
+      throw (
+        "See deployment.json: template is missing for " +
+        protocol +
+        " " +
+        network
+      );
+    }
+  }
+
+  // Checks if the network level data necessary to build and deploy this subgraph is present and/or valid.
+  checkNetworkLevelData(protocol, network) {
+    if (this.data[protocol]["networks"][network] == undefined) {
+      throw "Check --NETWORK spelling as argument or in deployment.json: network=network";
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()] ==
+      undefined
+    ) {
+      throw (
+        "See deployment.json: Service is missing for: protocol=" +
+        protocol +
+        " and network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "subgraph-slug"
+      ] == undefined
+    ) {
+      throw (
+        "See deployment.json: subgraph-slug is not defined for: protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "version"
+      ] == undefined
+    ) {
+      throw (
+        "version is not defined for protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network]["files"]["template"] == undefined
+    ) {
+      throw (
+        "See deployment.json: template is missing for: protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    }
+    this.checkValidVersion(
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "version"
+      ]
+    );
+  }
+
+  checkAuthorization() {
+    if (!this.access && this.getServiceByAlias() == "cronos-portal") {
+      throw "please specify an authorization token";
+    }
+  }
+
+  // Runs all checks for valid input data.
+  isValidInput() {
+    if (!this.target && this.type != "build") {
+      throw "Please specify a target location if you are deploying. If you are trying to build, set type=build";
+    }
+    if (!["build", "deploy", "check", ""].includes(this.type)) {
+      throw "Please specify a valid type: type=" + this.type;
+    }
+    if (this.type != "build") {
+      this.checkAuthorization();
+      this.checkValidService();
+    }
+    this.checkValidScope();
+  }
+
+  // This checks if the deployment json file is valid for the deployments that you want to execute.
+  isValidJsonData() {
+    let scope = this.getDeploymentScope();
+    if (scope == "network") {
+      this.checkProtocolLevelData(this.protocol);
+      if (this.type == "deploy" || this.type == "check") {
+        this.checkNetworkLevelData(this.protocol, this.network);
+      } else if (this.type == "build") {
+        this.checkNetworkLevelBuildDataOnly(this.protocol, this.network);
```

Comment:
This function looks like it can be simplified.

- You use this in every branch. Might move it to its own function.
- Replace the elses with early returns.
- Calls to `this.checkProtocolLevelData(this.protocol);` are in all branches but the forks one. You might try to generalize it too.

Reply:
What do you mean by generalizing on the last point? Protocol level data is still checked when the scope is 'forks'.  By generalizing, do you mean combining the protocol and network level data checks?

Reply:
Oh disregard the last point. Looking at it now it might not make much sense. Looks way better already, so not necessary either :D